pkg/config/lima_config_applier.go

```diff
@@ -107,9 +111,65 @@ func NewLimaApplier(
 		limaDefaultConfigPath:  limaDefaultConfigPath,
 		limaOverrideConfigPath: limaOverrideConfigPath,
 		systemDeps:             systemDeps,
+		finchConfigPath:        finchConfigPath,
 	}
 }
 
+func loadFinchConfig(fs afero.Fs, finchConfigPath string, logger flog.Logger, systemDeps LoadSystemDeps, mem fmemory.Memory) (*Finch, error) {
+	b, err := afero.ReadFile(fs, finchConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read finch.yaml: %w", err)
+	}
+
+	var cfg Finch
+	if err := yaml.Unmarshal(b, &cfg); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config file: %w", err)
+	}
+	if err := validate(&cfg, logger, systemDeps, mem); err != nil {
+		return nil, fmt.Errorf("failed to validate config file: %w", err)
+	}
+
+	return &cfg, nil
+}
+
+func (lca *limaConfigApplier) ModifyFinchConfig(fs afero.Fs, logger flog.Logger, cpus int, memory string) (bool, error) {
+	var isConfigUpdated bool
+
+	systemDeps := system.NewStdLib()
+	mem := fmemory.NewMemory()
+
+	finchCfg, err := loadFinchConfig(fs, lca.finchConfigPath, logger, systemDeps, mem)
+	if err != nil {
+		return isConfigUpdated, fmt.Errorf("failed to load config: %w", err)
+	}
+
+	if cpus == 0 && memory == "" {
+		return isConfigUpdated, fmt.Errorf("specified number of CPUs or specified amount of memory should be valid values")
+	}
+
+	if cpus > 0 && *finchCfg.CPUs != cpus {
```

Comment:
if cpus > 0 check should be part of validation in the above check

Reply:
OK, I'll fix it.

Reply:
Fixed logic for validation.